forem/forem
# Add a contact email address to the /admin/customization/config

## What type of PR is this? (check all applicable)


## Description
This PR introduces a contact_link. 

Currently across the site, we use the `default_email` as the email. This email displays the email address from which emails are being sent from for Forem Cloud instances that are configured with Sendgrid. 

However, this `default_email` is also being used for contact links across the site. As an example it will say:
"Contact [noreply@forem.com](mailto:noreply@forem.com?subject=Custom%20Bulk%20Pricing%2FPartnerships) for custom bulk pricing and partnerships."

Hence, I've added a specific contact field. The contact field gives the Forem an opportunity to provide an email that are used for any contact links. It is the email where users can get in touch with the Forem or their team."

In doing so, I took the opportunity to clean up the user interface and experience a little bit. 
- I've removed the "email Digest Frequency Section" and moved the "periodic email digest" field to the "Emails Section"
- I've added the contact field to the email section as well
- Finally, I've removed the "Default" field form the Emails section. Instead we've added a note in this [PR](https://github.com/forem/forem/pull/16223/files) that states "As a Forem Cloud client, we provide an email server managed by the Forem team. All settings are managed by us and the from and reply email addresses are set as [noreply@forem.com](mailto:noreply@forem.com). However, you can override this to use your own email server." which serves a similar but more informative purpose.

BEFORE: 
<img width="958" alt="Screenshot 2022-02-09 at 14 50 41" src="https://user-images.githubusercontent.com/2786819/153204604-8bb01982-e796-4a78-94bb-d9fc413f36a2.png">

AFTER:
<img width="1070" alt="Screenshot 2022-02-09 at 14 53 05" src="https://user-images.githubusercontent.com/2786819/153205054-8160975f-0661-4d6f-bd59-29a16f7a4a87.png">

I want to make a note that the default value for the contact field will however be the "default email" that is used for Sendgrid emails. This is so that views render properly with a value on first load. 

https://www.loom.com/share/e1615e5442d5422382edb5db603ead43

## Related Tickets & Documents
https://github.com/forem/forem/issues/16180

## QA Instructions, Screenshots, Recordings

1. Make sure that you have a `DEFAULT_EMAIL` value set in your env
2. Go to `http://localhost:3000/admin/customization/config` - Email Section, and see that the default value for the `contact email` is set to the `DEFAUL_EMAIL` value.
3. You should see this email on http://localhost:3000/credits/new
<img width="1263" alt="Screenshot 2022-02-09 at 14 56 20" src="https://user-images.githubusercontent.com/2786819/153205807-ec71e6a7-7ad0-435c-b6f4-d36d38cfe47c.png">
4. Now go back to `http://localhost:3000/admin/customization/config` - Email Section, and update the contact email and see that it updates on /credits/new.


### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/MEjG5NXKKS68LvGIP2/giphy.gif)
